tests/components/template/test_fan.py

```diff
@@ -222,6 +222,64 @@ async def test_templates_with_entities(hass, calls):
     _verify(hass, STATE_ON, SPEED_MEDIUM, True, DIRECTION_FORWARD)
 
 
+async def test_template_with_unavailable_entities(hass, calls):
+    """Test unavailability with value_template."""
+
+    with assert_setup_component(1, "fan"):
+        assert await setup.async_setup_component(
+            hass,
+            "fan",
+            {
+                "fan": {
+                    "platform": "template",
+                    "fans": {
+                        "test_fan": {
+                            "value_template": "{{ 'unavailable' }}",
+                            "turn_on": {"service": "script.fan_on"},
+                            "turn_off": {"service": "script.fan_off"},
+                        }
+                    },
+                }
+            },
+        )
+
+    await hass.async_block_till_done()
+    await hass.async_start()
+    await hass.async_block_till_done()
+    assert hass.states.get(_TEST_FAN).state == STATE_OFF
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=5872960)